iron-list.html

```diff
@@ -1870,15 +1872,31 @@
         }
       }
     },
-
-    _didMoveUp: function() {
-      this._focusPhysicalItem(this._focusedIndex - 1);
+    // When `grid === true` the `downArrow` goes forward `this._itemsPerRow` items
+    _didMoveBackward: function(e) {;
+      this._focusPhysicalItem(this._focusedIndex - (!this.grid ? 1 : this._itemsPerRow));
     },
-
-    _didMoveDown: function(e) {
+    // When `grid === true` the `upArrow` goes backward `this._itemsPerRow` items
+    _didMoveForward: function(e) {;
+      var destination = this._focusedIndex + (!this.grid ? 1 : this._itemsPerRow);
+	  // When `grid === true` move focus down one row, either
+	  // directly below or the last item of the list.
+	  if (this.grid && this._focusedIndex < (this.items.length - (this.items.length % this._itemsPerRow))) {
+		  destination = Math.min(destination, this.items.length - 1);
+	  }
```

Comment:
It's unclear to me how the use of `_focusPhysicalItem(this._focusedIndex + (this.grid ? this._itemsPerRow : 1))` allows for both of the cases that I've outlined above to be true. My current goal is for `_didMoveForward` to NOT work exactly like `_didMoveBackward` in that moving forward can intersect with an incomplete row while moving backwards cannot.

If there is some concept of code simplicity or API contact requirements that prevent that from being true, I can completely understand. Just let me know that's the case, and my UX team and I can agree on a simpler, unified mechanism of moving forward and backward to submit to this PR.